lib/org/chromium/webidl/IDLFragmentExtractorRunner.js

```diff
@@ -0,0 +1,46 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+'use strict';
+
+foam.CLASS({
+  package: 'org.chromium.webidl',
+  name: 'IDLFragmentExtractorRunner',
+  extends: 'org.chromium.webidl.PipelineRunner',
+
+  requires: [
+    'org.chromium.webidl.PipelineMessage',
+    'org.chromium.webidl.IDLFragmentExtractor as Extractor',
+    'org.chromium.webidl.IDLFileContents',
+  ],
+
+  documentation: 'Runnable Box that extracts IDL Fragments from HTMLFileContents.',
+
+  methods: [
+    function run(message) {
+      var sup = this.SUPER(message);
```

Comment:
The pattern introduces a new contract. `run()` generally isn't expected to return a value. We should, at least, add documentation to `PipelineRunner` to call this out. It may be better, though, to create a different method responsible for input error checking, and invoke it explicitly.